test/OpenTelemetry.Tests/Metrics/MetricApiTestsBase.cs

```diff
@@ -1429,7 +1431,7 @@ int MetricPointCount()
         }
 
         meterProvider.ForceFlush(MaxTimeToAllowForFlush);
-        Assert.Equal(MeterProviderBuilderSdk.DefaultCardinalityLimit, MetricPointCount());
+        Assert.Equal(MeterProviderBuilderSdk.DefaultCardinalityLimit + additionalReserve, MetricPointCount());
```

Comment:
This test should verify that we don't allow the user to exceed the cardinality cap. The right thing to test here now would be to check that count of metric points exported (excluding zero tags and overflow attribute) is equal to `MeterProviderBuilderSdk.DefaultCardinalityLimit`. We probably need to update `MetricPointCount()` method to return the count of unreserved MetricPoints instead of all the MetricPoints that were exported in that particular Collect cycle.

Reply:
The overflow is guarded with an experimental flag. In cases where the overflow experimental attribute is not set, the value will be the cardinality limit + 1.

Reply:
It looks like you have now updated `MetricPointCount()` method definition to account for that.